heron/schedulers/src/java/com/twitter/heron/scheduler/aurora/AuroraCLIController.java

```diff
@@ -111,29 +112,33 @@ public void removeContainers(Set<PackingPlan.ContainerPlan> containersToRemove)
   }
 
   @Override
-  public void addContainers(Integer count) {
+  public Set<Integer> addContainers(Integer count) {
     //aurora job add <cluster>/<role>/<env>/<name>/<instance_id> <count>
     //clone instance 0
     List<String> auroraCmd = new ArrayList<>(Arrays.asList(
-        "aurora", "job", "add", "--wait-until", "RUNNING", jobSpec + "/0", count.toString()));
-
-    if (isVerbose) {
-      auroraCmd.add("--verbose");
-    }
+        "aurora", "job", "add", "--wait-until", "RUNNING",
+        jobSpec + "/0", count.toString(), "--verbose"));
 
     LOG.info(String.format("Requesting %s new aurora containers %s", count, auroraCmd));
-    if (!runProcess(auroraCmd)) {
+    StringBuilder stderr = new StringBuilder();
+    if (!runProcess(auroraCmd, null, stderr)) {
       throw new RuntimeException("Failed to create " + count + " new aurora instances");
     }
+
+    String stdoutStr = stderr.toString();
+    String pattern = "Querying instance statuses: [";
+    int idx1 = stdoutStr.indexOf(pattern) + pattern.length();
+    int idx2 = stdoutStr.indexOf("]", idx1);
+    String containerIdStr = stdoutStr.substring(idx1, idx2);
+    LOG.info("container IDs returned by Aurora " + containerIdStr);
+    return Arrays.asList(containerIdStr.split(", "))
+        .stream().map(x->Integer.valueOf(x)).collect(Collectors.toSet());
   }
 
-  // Utils method for unit tests
-  @VisibleForTesting
-  boolean runProcess(List<String> auroraCmd) {
-    StringBuilder stdout = new StringBuilder();
-    StringBuilder stderr = new StringBuilder();
+  boolean runProcess(List<String> auroraCmd, StringBuilder stdout, StringBuilder stderr) {
```

Comment:
should this function be VisibleForTesting too? Seems more suitable for unit tests.

Reply:
updated with test